auth/client/iam/client.go

```diff
@@ -120,3 +128,47 @@ func (hb HTTPClient) PresentationDefinition(ctx context.Context, definitionEndpo
 
 	return &presentationDefinition, nil
 }
+
+func (hb HTTPClient) AccessToken(ctx context.Context, tokenEndpoint string, vp vc.VerifiablePresentation, submission pe.PresentationSubmission, scopes string) (oauth.TokenResponse, error) {
+	var token oauth.TokenResponse
+	presentationDefinitionURL, err := url.Parse(tokenEndpoint)
+	if err != nil {
+		return token, err
+	}
+
+	// create a POST request with x-www-form-urlencoded body
+	assertion, _ := json.Marshal(vp)
+	presentationSubmission, _ := json.Marshal(submission)
+	data := url.Values{}
+	data.Set("grant_type", "vp_token-bearer")
+	data.Set("assertion", string(assertion))
+	data.Set("presentation_submission", string(presentationSubmission))
+	data.Set("scope", scopes)
+	request, err := http.NewRequestWithContext(ctx, http.MethodPost, presentationDefinitionURL.String(), strings.NewReader(data.Encode()))
+	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	if err != nil {
+		return token, err
+	}
+	response, err := hb.httpClient.Do(request.WithContext(ctx))
+	if err != nil {
+		return token, fmt.Errorf("failed to call endpoint: %w", err)
+	}
+	if err = core.TestResponseCode(http.StatusOK, response); err != nil {
+		// check for oauth error
+		if innerErr := core.TestResponseCode(http.StatusBadRequest, response); innerErr == nil {
+			// a non oauth error, the response body could contain a lot of stuff. We'll log and return the entire error
+			log.Logger().Errorf("authorization server token endpoint returned non oauth error (statusCode=%d)", response.StatusCode)
+		}
+
+		return token, err
+	}
+
+	var responseData []byte
+	if responseData, err = io.ReadAll(response.Body); err != nil {
+		return token, fmt.Errorf("unable to read response: %w", err)
+	}
+	if err = json.Unmarshal(responseData, &token); err != nil {
+		return token, fmt.Errorf("unable to unmarshal response: %w, %s", err, string(responseData))
```

Comment:
might be very long string (e.g. HTML page), so clip it or don't include it imo (or  log on debug?)

Reply:
clipped it accordingly with v1 client